src/app.rs

```diff
@@ -155,14 +156,94 @@ impl<'a, 'b> Application<'a, 'b> {
             #[cfg(feature = "profiler")]
             profile_scope!("handle_event");
 
-            self.events.poll_events(|event| {
-                if let Some(mut event_handler) =
-                    engine.world.res.try_fetch_mut::<EventHandler<Event>>(0)
-                {
+            // Input devices can sometimes generate a lot of motion events per frame, these are
+            // useless as the extra precision is wasted and these events tend to overflow our
+            // otherwise very adequate event buffers.  So we're going to only keep the most recent
+            // motion axis event per axis, per device, per frame.
+
+            // Allocate a place to store events, we need this so we can compare events
+            // against previous events and determine if we need to discard older events
+            //
+            // NOTE: Type is explicit here to make sure we're not pushing references in later code.
+            let mut events: Vec<Event> = Vec::new();
+            self.events.poll_events(|new_event| {
+                // This boolean helps us avoid writing the same piece of code over and over.
+                let mut write_new_event = true;
+                match new_event {
+                    Event::WindowEvent { ref event, .. } => match event {
+                        &WindowEvent::MouseMoved { .. } => for mut stored_event in &mut events {
```

Comment:
You're saying that events are too many, but with this we have to iterate over all so far collected events *for every new event*. Is that a good thing?

Reply:
No, you don't iterate over them. Because you don't push them here in the first place.

Reply:
This vec should be kept small, the intent is that this vec contains all the events we write to the buffer, so instead of writing thousands we're now only writing one per input axis.